Add Wreck support for version 11 and above

lostthetrail opened the pull request:
Resolves #549

renegare commented on test/monitor.js:
```diff
@@ -1018,7 +1027,7 @@ describe('Monitor', () => {
                         expect(res.statusCode).to.equal(200);
                         // account for hapi15 differences
-                        expect(out.data.length).to.be.greaterThan(0);
+                        expect(out.data.length).to.be.greaterThan(1);
```
@lostthetrail I think this should be set to `least`. However; thats just to make it pass. Not entirely sure why one can't assert one event given you only make one wreck request in this single test case 🤔

lostthetrail replied:
Drat. I had changed it from 0 to 1, when I was originally adjusting the unit test for wreck to confirm both cases (before I split it into two tests). I will revert that change in the next hour. Thanks!

arb commented on lib/monitor.js:
```diff
@@ -68,6 +68,15 @@ class Monitor {
             this._wreckHandler = (error, request, response, start, uri) => {
 
+                // Wreck 11+ emits response event with signature (err, {req, res, start, uri})
+                const isWreckVersion11OrHigher = (!response && typeof request === 'object');
```
Thoughts on making this a proper `function` and checked `arguments.length` instead?

arb commented:
CI failing.

lostthetrail commented:
@arb Since this was a bug fix more than a new feature enhancement, I changed as little as possible.

Would you prefer if I adjusted from arrow to function and use the arguments list to dynamically determine values?

arb commented:
Yeah I think so.

lostthetrail commented:
You got it. Standby.

lostthetrail commented:
@arb Back to you. Anything else you would like adjusted?
